pilot/pkg/networking/core/cluster_traffic_policy.go

```diff
@@ -332,12 +350,39 @@ func applyLeastRequestLoadBalancer(c *cluster.Cluster, loadbalancer *networking.
 }
 
 // setSlowStartConfig will set the warmupDurationSecs for LEAST_REQUEST and ROUND_ROBIN if provided in DestinationRule
+// Deprecated: use setWarmup instead
 func setSlowStartConfig(dur *durationpb.Duration) *cluster.Cluster_SlowStartConfig {
 	return &cluster.Cluster_SlowStartConfig{
 		SlowStartWindow: dur,
 	}
 }
 
+// setWarmup will set the warmup configuration for LEAST_REQUEST and ROUND_ROBIN if provided in DestinationRule
+// Runtime key is not exposed,the value is fixed
+func setWarmup(warmup *networking.WarmupConfiguration) *cluster.Cluster_SlowStartConfig {
+	var aggression, minWeightPercent float64
+
+	// If not specified, aggression defaults to 1.0 ensuring a linear traffic rampup
+	if a := warmup.Aggression; a == nil {
+		aggression = 1
+	} else {
+		aggression = a.GetValue()
+	}
+
+	// If not specified, minWeightPersent default to 10, aligned with envoy default
+	if m := warmup.MinimumPercent; m == nil {
+		minWeightPercent = 10
+	} else {
+		minWeightPercent = m.GetValue()
+	}
+
+	return &cluster.Cluster_SlowStartConfig{
+		SlowStartWindow:  warmup.Duration,
+		Aggression:       &core.RuntimeDouble{DefaultValue: aggression, RuntimeKey: "("},
```

Comment:
what does "(" means ?

Reply:
It's the runtime key : any character/string is valid and used by envoy to get value for calculating the percentage of requests. This is not exposed to the user, that's why it's hardcoded with "("

Reply:
ok